src/components/SwitchButton/index.tsx

```diff
@@ -1,204 +1,141 @@
-    }
+  &[data-position='right']:active:before {
+    width: calc(50% + ${({ theme }) => theme.space['0.5']});
   }
 
-  &:not([data-checked='true']):not([aria-disabled='true']) {
-    &:hover,
-    &:focus {
-      border-color: ${({ theme }) => theme.colors.primary.border};
-    }
+  &[data-position='left']:active:before {
+    width: calc(50% - ${({ theme }) => theme.space['0.5']} - 1px);
+  }
+`
 
-    &:hover {
-      box-shadow: ${({ theme }) => theme.shadows.hoverPrimary};
-    }
+const StyledSelectableCard = styled(SelectableCard)<{ checked: boolean }>`
+  border: none;
+  height: 40px;
+  padding: ${({ theme }) => theme.space['1']} ${({ theme }) => theme.space['2']};
+  font-weight: ${({ theme }) => theme.typography.bodyStrong.weight};
+  justify-content: center;
+  align-items: center;
 
-    &:focus {
-      box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
-    }
+  &:hover,
+  &:focus-within,
+  &:active {
+    box-shadow: none;
+    border: none;
   }
 
-  ${({ theme }) => Object.values(variants).map(variantFn => variantFn(theme))}
-`
-
-const StyledRadio = styled(Radio)`
-  position: absolute;
-  opacity: 0.01;
+  ${({ checked, theme }) =>
+    checked
+      ? `
+  color: ${theme.colors.primary.textStrong};
+  `
+      : `
+        &:hover {
+          color: ${theme.colors.primary.textWeak};
+        }
+      `}
 `
 
-type SwitchButtonProps = Omit<StyledSwitchProps, 'onChange' | 'children'> & {
-  children:
-    | ReactNode
-    | (({
-        checked,
-        disabled,
-      }: {
-        checked: boolean
-        disabled: boolean
-      }) => ReactNode)
+type SwitchButtonProps = {
   name: string
   onBlur?: FocusEventHandler
   onChange: ChangeEventHandler
   onFocus?: FocusEventHandler
   tooltip?: string
-  value: string
-  checked?: boolean
-  disabled?: boolean
-  variant?: Variants
+  value?: string | number
+  leftValue: string | number
+  rightValue: string | number
+  leftText: string
+  rightText: string
 }
 
 const SwitchButton = ({
-  checked = false,
-  disabled = false,
+  value,
   onChange,
   onFocus,
   onBlur,
   name,
-  value,
-  children,
+  leftValue,
+  rightValue,
+  leftText,
+  rightText,
   tooltip,
-  variant,
-  ...props
-}: SwitchButtonProps) => (
-  <Tooltip text={tooltip}>
-    <StyledSwitch
-      as="label"
-      htmlFor={`${name}-${value}`}
-      data-checked={checked}
-      aria-disabled={disabled}
-      data-variant={variant}
-      {...props}
-    >
-      {typeof children === 'function'
-        ? children({ checked, disabled })
-        : children}
-      <StyledRadio
-        checked={checked}
-        disabled={disabled}
-        id={`${name}-${value}`}
-        name={name}
-        onBlur={onBlur}
-        onChange={onChange}
-        onFocus={onFocus}
-        type="radio"
-        value={value}
-      />
-    </StyledSwitch>
-  </Tooltip>
-)
+}: SwitchButtonProps) => {
+  const [localValue, setLocalValue] = useState(
+    value === 'left' ? leftValue : rightValue,
```

Comment:
Shouldn't we check `if (value === leftValue)` instead?

Reply:
@QuiiBz You speak truth